packages/genesys-spark-components/src/components/stable/gux-form-field/components/gux-form-field-file-beta/gux-form-field-file.tsx

```diff
@@ -0,0 +1,358 @@
+import {
+  Component,
+  State,
+  Element,
+  JSX,
+  Prop,
+  h,
+  Listen,
+  forceUpdate
+} from '@stencil/core';
+
+import { buildI18nForComponent, GetI18nValue } from '../../../../../i18n';
+import { ILocalizedComponentResources } from '../../../../../i18n/fetchResources';
+
+import { OnMutation } from '@utils/decorator/on-mutation';
+import { hasSlot } from '@utils/dom/has-slot';
+import { trackComponent } from '@utils/tracking/usage';
+import {
+  GuxFormFieldError,
+  GuxFormFieldContainer,
+  GuxFormFieldHelp,
+  GuxFormFieldLabel
+} from '../../functional-components/functional-components';
+import { GuxFormFieldIndicatorMark } from '../../gux-form-field.types';
+import { getSlottedInput, validateFormIds } from '../../gux-form-field.service';
+import { preventBrowserValidationStyling } from '@utils/dom/prevent-browser-validation-styling';
+import {
+  onDisabledChange,
+  onRequiredChange
+} from '@utils/dom/on-attribute-change';
+
+import componentResources from './i18n/en.json';
+
+/**
+ * @slot input - Required slot for input tag
+ * @slot label - Required slot for label tag
+ * @slot error - Optional slot for error message
+ * @slot help - Optional slot for help message
+ * @slot label-info - Optional slot for tooltip
+ */
+
+@Component({
+  styleUrl: 'gux-form-field-file.scss',
+  tag: 'gux-form-field-file-beta',
+  shadow: true
+})
+export class GuxFormFieldFileBeta {
+  private getI18nValue: GetI18nValue;
+  private dropContainer: HTMLDivElement;
+  private input: HTMLInputElement;
+  private labelInfo: HTMLGuxLabelInfoBetaElement;
+  private disabledObserver: MutationObserver;
+  private requiredObserver: MutationObserver;
+  private hideLabelInfoTimeout: ReturnType<typeof setTimeout>;
+
+  @Element()
+  private root: HTMLElement;
+
+  /**
+   * Field indicator mark which can show *, (optional) or blank
+   * Defaults to required. When set to required, the component will display * for required fields and blank for optional
+   * When set to optional, the component will display (optional) for optional and blank for required.
+   */
+  @Prop()
+  indicatorMark: GuxFormFieldIndicatorMark = 'required';
+
+  @Prop()
+  dropAndDrag: boolean = false;
+
+  @State()
+  private disabled: boolean = false;
+
+  @State()
+  private hasError: boolean = false;
+
+  @State()
+  private hasHelp: boolean = false;
+
+  @State()
+  private required: boolean = false;
+
+  @OnMutation({ childList: true, subtree: true })
+  onMutation(): void {
+    this.labelInfo = this.root.querySelector('[slot=label-info]');
+    this.hasError = hasSlot(this.root, 'error');
+    this.hasHelp = hasSlot(this.root, 'help');
+  }
+
+  @Listen('keyup')
+  handleKeyup(event: KeyboardEvent): void {
+    switch (event.key) {
+      case 'Tab': {
+        if (this.input.matches(':focus-visible')) {
+          void this.labelInfo?.showTooltip();
```

Comment:
When I was reviewing the component example I found that the label tooltip becoming visible when the file upload button was focused seemed kind of strange just because of the distance between the button and the label on the example. But I don't think there's anything technically wrong with it

Reply:
ya the same functionality is in the other form field components so I will leave it here unless UX wants it changed